core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -137,6 +138,22 @@ trait EntityDecoderInstances {
     EntityDecoder.decodeBy(MediaRange.`*/*`)(collectBinary)
   }
 
+  implicit def sourceDecoder[A](implicit W: EntityDecoder[A]): EntityDecoder[Process[Task,A]] =
+    new EntityDecoder[Process[Task,A]] {
+      override def decode(msg: Message): DecodeResult[Process[Task,A]] = {
+        val result = msg.body.flatMap{ byteVector =>
+          val fakeMessage = Request(body = Process.emit(byteVector))
+          Process.await(W.decode(fakeMessage).run) {
```

Comment:
I think what @rossabaker is trying to say is that each 'chunk' (in the form of a `ByteVector`) doesn't necessarily correspond to any decoded representation. This applies at both the `EntityBody` level, and the [http Transfer-Encoding level](https://tools.ietf.org/html/rfc7230#section-3.3.1).

Basically, this setup will only work if the chunks you receive correspond to one and only one `A`, and this must not be expected per the http specification. You need a way to accumulate 'chunks' until your decoder has what it needs for a full `A` and save any leftover for the next `A` coming off the wire.